File: output/gemma4-e4b/avatar/Python/Java/atcoder_ABC168_C.java
import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC168_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read a, b, h, m from standard input
        if (!scanner.hasNextInt()) {
            return;
        }
        
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int h = scanner.nextInt();
        int m = scanner.nextInt();
        
        scanner.close();
        
        // Calculate C: C = abs(360/60 * m - 360/12 * h - 360/12/60 * m)
        // Using double for floating point arithmetic
        double c = Math.abs(6.0 * m - 30.0 * h - 0.5 * m);
        
        // Angle adjustment
        if (c > 180) {
            c = 360.0 - c;
        }
        
        // Calculate cosC
        double cosC = Math.cos(Math.toRadians(c));
        
        // Calculate the final result using the Law of Cosines
        double result = Math.sqrt(a * a + b * b - 2 * a * b * cosC);
        
        System.out.println(result);
    }
}

File: output/gemma4-e4b/avatar/Java/Python/atcoder_ABC168_C.py
import math
import sys

def solve():
    # Read all four integers from standard input
    try:
        data = sys.stdin.read().split()
        if len(data) < 4:
            return
        A = int(data[0])
        B = int(data[1])
        H = int(data[2])
        M = int(data[3])
    except Exception:
        return

    # 1. Calculate angle contributions (in degrees)
    # Minutes contribution: M * 6 degrees
    mkaku = M * 6.0
    
    # Hours contribution: H * 30 + M * 0.5 degrees
    hkaku = H * 30.0 + M * 0.5
    
    # Angle difference (kaku)
    kaku = mkaku - hkaku
    
    # 2. Convert absolute angle difference to radians
    d_kaku = math.radians(abs(kaku))
    
    # 3. Calculate ans2 = (A^2 + B^2) - 2 * A * B * cos(d_kaku)
    
    # CC = A^2 + B^2
    cc = A*A + B*B
    
    # DD = 2 * A * B * cos(d_kaku)
    dd = 2.0 * A * B * math.cos(d_kaku)
    
    ans2 = cc - dd
    
    # 4. Final result: sqrt(ans2)
    ans = math.sqrt(ans2)
    
    print(ans)

if __name__ == "__main__":
    solve()